apps/fz_http/lib/fz_http/api_tokens.ex

```diff
@@ -0,0 +1,53 @@
+defmodule FzHttp.ApiTokens do
+  @moduledoc """
+  The ApiTokens context.
+  """
+
+  import Ecto.Query, warn: false
+  alias FzHttp.Repo
+
+  alias FzHttp.ApiTokens.ApiToken
+
+  def count, do: count(ApiToken)
+  def count(nil), do: nil
+
+  def count(queryable) when is_struct(queryable) or is_atom(queryable),
+    do: Repo.aggregate(queryable, :count)
+
+  def count(user_id), do: count(from(a in ApiToken, where: a.user_id == ^user_id))
+
+  def list_api_tokens do
+    Repo.all(ApiToken)
+  end
+
+  def list_api_tokens(user_id) do
+    Repo.all(from a in ApiToken, where: a.user_id == ^user_id)
+  end
+
+  def get_api_token(id), do: Repo.get(ApiToken, id)
+
+  def get_api_token!(id), do: Repo.get!(ApiToken, id)
+
+  def new_api_token(attrs \\ %{}) do
+    ApiToken.changeset(%ApiToken{}, attrs)
+  end
+
+  def create_api_token(attrs \\ %{}) do
+    FzHttp.Telemetry.create_api_token()
+    user_id = attrs[:user_id] || attrs["user_id"]
+
+    %ApiToken{}
+    |> ApiToken.changeset(attrs, count_per_user: count(user_id))
+    |> Repo.insert()
+  end
```

Comment:
So there are a few things here:
1. I feel like passing `%User{}` struct as an argument would result in a much more explicit code. Right now you kinda need to guess that you must set the assoc by passing a "user_id" attribute, it even forces you to add a hack to read it because params can be atom- or string-keyed;
2. We might want to track telemetry only on successful actions, otherwise you can spam it heavily by rapid invalid API requests;
3. Do we really want to enforce the count limit per user? People that create those tokens are admins, I suppose they know what they are doing, but having just 10 hardcoded tokens might be too small for some of their use cases.
4. Default changeset that you generate by `mix phx.gen.context` forces you to pass an empty struct but IRL you very rarely want once changeset for creating and updating (eg. token should not be updated at all). I would add a separate function per action to the changeset and make sure create doesn't accept anything (always starts from a clean struct).

```suggestion
  def create_api_token(attrs \\ %{}) do
    user_id = attrs[:user_id] || attrs["user_id"]
    changeset = ApiToken.create_changeset(attrs, count_per_user: count_by_user_id(user_id))

    with {:ok, api_token} <- Repo.insert(changeset) do
      FzHttp.Telemetry.create_api_token()
      {:ok, api_token)
    end
```

Reply:
In agreement with (1), (2), and (4). For (3) I was thinking it makes sense to have *some* sane limit, for two reasons:
1. Beyond $LIMIT records, the table and page become unusable (see pagination backlog issue).
2. If an admin is creating more than $LIMIT api tokens, might it be a sign of a mismatch between how we've designed this feature and how they're using it? I would love to learn more about their use-case -- I can't imagine many cases where that many tokens is warranted. Maybe the error message can prompt them to chat with us 😈.

Reply:
Updated this to `25`.

Reply:
Perfect. And we will handle the pagination a bit later.